core/data/permission/ReverseRightLookupInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\generis\model\data\permission;
+
+/**
+ * ReverseRightLookupInterface
+ *
+ * @package oat\generis\model\data\permission
+ */
+interface ReverseRightLookupInterface
+{
+    /**
+     * Returns a list roles and permissions related to a resource
+     *
+     *
+     * Sample data to be returned:
+     * [
+     *    'http://www.tao.lu/Ontologies/TAO.rdf#BackOfficeRole' => [
+     *        'GRANT',
+     *        'READ',
+     *        'WRITE'
+     *    ],
+     *    'http://www.tao.lu/Ontologies/TAO.rdf#ItemAuthor' => [
+     *        'GRANT',
+     *        'READ'
+     *    ],
+     * ]
```

Comment:
It's better to extract those into a `PermissionCollection` entity.

Reply:
> Don't forget the `manifest.php`.

It won't be done right now but you are right, we will check it when merging `feature/TAO-10203-advanced-search` to `develop` 

> It's better to extract those into a `PermissionCollection` entity.
> Without it, the interface doesn't really enforce any contract :)

I totally agree, kind o hate these arrays everywhere but unfortunately it's being used in many places, so I can't change it right now (not the focus here)